packages/kbar/src/hooks/useRefineKbar/index.tsx

```diff
@@ -0,0 +1,268 @@
+import { useEffect, useState, useContext } from "react";
+import {
+    ResourceRouterParams,
+    useRouterContext,
+    useNavigation,
+    useDelete,
+    useTranslate,
+    useResource,
+    IResourceItem,
+    useCanWithoutCache,
+} from "@pankod/refine-core";
+import {
+    useRegisterActions,
+    createAction,
+    Action,
+    KBarContext,
+    VisualState,
+} from "kbar";
+
+import { capitalize } from "@definitions";
+
+enum RefineKbarActionType {
+    List = "list",
+    Create = "create",
+    Show = "show",
+    Edit = "edit",
+    Delete = "delete",
+}
+
+export const useRefineKbar = (): void => {
+    const t = useTranslate();
+    const { resources } = useResource();
+    const { useParams } = useRouterContext();
+    const { mutate } = useDelete();
+    const kbarContext = useContext(KBarContext);
+
+    const { can } = useCanWithoutCache();
+
+    const {
+        resource: resourceFromRoute,
+        action: actionFromRoute,
+        id: idFromRoute,
+    } = useParams<ResourceRouterParams>();
+
+    const [actions, setActions] = useState<Action[]>([]);
+
+    const {
+        list: goToList,
+        create: goToCreate,
+        show: goToShow,
+        edit: goToEdit,
+    } = useNavigation();
+
+    useEffect(() => {
+        const preaparedActions = async () => {
+            return await Promise.all(
+                moveActionToFirst().flatMap((resource) => {
+                    return createActionWithResource(resource);
+                }),
+            );
+        };
+
+        preaparedActions().then((actions) => {
+            return setActions(actions.flatMap((action) => action));
+        });
+    }, [resources, idFromRoute, resourceFromRoute, actionFromRoute]);
+
+    useEffect(() => {
+        if (actions.length === 0) {
+            kbarContext.query.setVisualState(VisualState.hidden);
+        }
+    }, [actions]);
+
+    const moveActionToFirst = (): IResourceItem[] => {
+        const orderedResources = [...resources];
+        const fromIndex = orderedResources?.findIndex(
+            (resource) => resource.name === resourceFromRoute,
+        );
+
+        if (fromIndex > 0) {
+            const element = orderedResources[fromIndex];
+            orderedResources.splice(fromIndex, 1);
+            orderedResources.splice(0, 0, element);
+        }
+
+        return orderedResources;
+    };
+
+    const createActionWithResource = async (resource: IResourceItem) => {
+        const {
+            name,
+            label,
+            list,
+            create,
+            canCreate,
+            canEdit,
+            canShow,
+            icon,
+            show,
+            canDelete,
+            edit,
+            route,
+        } = resource;
+
+        const resourceName = label ?? name;
+        const tempActions: Action[] = [];
+        const section = t(`${resourceName}.${resourceName}`, resourceName);
```

Comment:
```suggestion
        const resourceName = label ?? name;
        const tempActions: Action[] = [];
        const section = t(`${resourceName}.${resourceName}`, resourceName);
```
We must handle `resource titles` like `useMenu` hook,
https://github.com/pankod/refine/blob/next/packages/core/src/hooks/menu/useMenu.tsx#L68